src/transaction/validation.rs

```diff
@@ -1,42 +1,55 @@
+use super::{Input, Output, Transaction};
+
+#[cfg(feature = "std")]
+use super::Witness;
+
+#[cfg(feature = "std")]
 use crate::consts::*;
-use crate::{Input, Output, Transaction, Witness};
 
-use fuel_types::{Color, Word};
+#[cfg(feature = "std")]
+use fuel_types::{Bytes32, Color, Word};
 
 mod error;
 
 pub use error::ValidationError;
 
 impl Input {
+    #[cfg(feature = "std")]
     pub fn validate(
         &self,
+        txhash: &Bytes32,
         index: usize,
         outputs: &[Output],
         witnesses: &[Witness],
+        check_input_signature: bool,
     ) -> Result<(), ValidationError> {
         match self {
             Self::Coin { predicate, .. } if predicate.len() > MAX_PREDICATE_LENGTH as usize => {
                 Err(ValidationError::InputCoinPredicateLength { index })
             }
 
-            Self::Coin { predicate_data, .. }
-                if predicate_data.len() > MAX_PREDICATE_DATA_LENGTH as usize =>
-            {
+            Self::Coin { predicate_data, .. } if predicate_data.len() > MAX_PREDICATE_DATA_LENGTH as usize => {
```

Comment:
These `as usize` casts of `u64` feel weird to me, as if the constant is `> usize::MAX` it will silently truncate. If the constant is length of something, and often cast to `usize`, changing the type of the constant would seem cleaner.

Reply:
This is not introduced in this PR. However, the constants must be consistent with the runtime requirements.